aiohttp/http_parser.py

```diff
@@ -341,17 +344,24 @@ def get_content_length() -> Optional[int]:
                         if SEC_WEBSOCKET_KEY1 in msg.headers:
                             raise InvalidHeader(SEC_WEBSOCKET_KEY1)
 
-                        self._upgraded = msg.upgrade
-
                         method = getattr(msg, "method", self.method)
                         # code is only present on responses
                         code = getattr(msg, "code", 0)
 
+                        import pprint
+
+                        pprint.pprint(["upgraded", msg.upgrade, msg])
+                        self._upgraded = code == 101 and msg.upgrade
```

Comment:
c parser seems to only set this if code is 101

https://github.com/nodejs/llhttp/blob/8498ef9d8b0e9539c8c331cf59213529287789e1/src/native/http.c#L21